# UDACITY Demo - Triggering Lambda

In [3]:
import boto3
from sagemaker import get_execution_role ## The SageMaker role executing your notebook needs to have Lambda permissions. 
import json

client = boto3.client('lambda')

payload = {'key': 'value'}

# json.dumps turns a JSON-object-like python object into a string, and .encode('utf-8') encodes the 
# the string so that it can be properly passed to the client. 

payload_bytes = json.dumps(payload).encode('utf-8')   # Convert a 'json object' into byte array

response = client.invoke(                   # Invocation via boto3 session
    FunctionName='example123',
    InvocationType='Event',   # 'Event' Invocation: Cloudwatch (Events -> Rules)
    Payload=payload_bytes
)

In [4]:
print(response)

{'ResponseMetadata': {'RequestId': '570249ad-ac20-407b-a961-7f50c0ea95c7', 'HTTPStatusCode': 202, 'HTTPHeaders': {'date': 'Mon, 09 Jan 2023 12:05:48 GMT', 'content-length': '0', 'connection': 'keep-alive', 'x-amzn-requestid': '570249ad-ac20-407b-a961-7f50c0ea95c7', 'x-amzn-remapped-content-length': '0', 'x-amzn-trace-id': 'root=1-63bc031c-3a0c943f43fbc7884d3f1a0f;sampled=0'}, 'RetryAttempts': 0}, 'StatusCode': 202, 'Payload': <botocore.response.StreamingBody object at 0x7f57fab4f9d0>}


We see **200 (202) code** above, hence successful.

Go and check successful execution of 'example123' lambda function under <br>
*Lambda functions -> example123 (function passed here) -> monitor*